merino/middleware/geolocation.py

```diff
@@ -64,7 +60,7 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         except AddressNotFoundError:
             pass
 
-        ctxvar_geolocation.set(
+        scope[ScopeKey.GEOLOCATION] = (
```

Comment:
I like the idea of using an enum for this to avoid naming conflicts. 👍🏻 

This uses the enum member as the mapping key. I want to make sure that's what we want.

From what I can tell starlette's `Scope` expects `typing.MutableMapping[str, typing.Any]`. This is satisfied for `ScopeKey` since we use `str` as one of the base classes, but it may not be what we want. If want to use `merino_geolocation` as a key in scope, we need to retrieve the value first:

```suggestion
        scope[ScopeKey.GEOLOCATION.value] = (
```

Reply:
With it being a str subclass I'll argue using the enum variant itself doesn't matter much here as it all works just the same (maybe except you get an enum repr of these keys). Even `scope["merino_geolocation"]` works against the `Enum`.

However if you'd like I can just make `ScopeKey` a regular class of plain str constants instead of an enum.

Reply:
I think using an `Enum` is the right call here. I just wanted to make sure we're OK with using the enum member for the key rather than the string value.

I didn't know that `scope["merino_geolocation"]` works with that. Let's use an enum!

Reply:
👍, reverted back to the enum